components/Handlers.js

```diff
@@ -0,0 +1,45 @@
+import { render } from '@asyncapi/generator-react-sdk';
+import { pascalCase } from './common';
+
+
+let subscriptionFunction = (channelName, operation, message) => `
+//${operation} subscription handler for ${channelName}        
+func ${operation}(msg *message.Message) error {
+    log.Printf("received message payload: %s", string(msg.Payload))
+
+    var lm payloads.${message}
+    err := json.Unmarshal(msg.Payload, &lm)
+    if err != nil {
```

Comment:
From what I've tested, once you produce a message that is not JSON, this `err` will be returned and the consumer gets stuck reading the same message again and again. I think this is due to the fact you are returning an error, and as far as I read here https://watermill.io/docs/messages-router/#ack, ACK of the message is not happening.

Perhaps logging the json unmarshal error is enough, in particular for this generated code.

Reply:
yes makes sense addressed it